src/map.cpp

```diff
@@ -5205,6 +5206,52 @@ const trap &map::tr_at( const tripoint &p ) const
     return current_submap->get_trap( l ).obj();
 }
 
+partial_con *map::partial_con_at( const tripoint &p )
+{
+    if( !inbounds( p ) ) {
+        return nullptr;
+    }
+    point l;
+    submap *const current_submap = get_submap_at( p, l );
+    tripoint p_shifted( l.x, l.y, p.z );
+    auto it = current_submap->partial_constructions.find( p_shifted );
+    if( it != current_submap->partial_constructions.end() ) {
+        return &it->second;
+    }
+
+    return nullptr;
+}
+
+void map::partial_con_remove( const tripoint &p )
+{
+    if( !inbounds( p ) ) {
+        return;
+    }
+    point l;
+    submap *const current_submap = get_submap_at( p, l );
+    tripoint p_shifted( l.x, l.y, p.z );
+    if( partial_con_at( p ) != nullptr ) {
```

Comment:
I think this can be rewritten as simply
```cpp
    current_submap->partial_constructions.erase( tripoint( l.x, l.y, p.z ) );
```
If the key doesn't exist nothing happens, so there's no need to check for it first.